src/main/java/ch/njol/skript/expressions/ExprFacing.java

```diff
@@ -89,36 +98,37 @@ public Class<Direction> getReturnType() {
 	
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
+	public Class<?>[] acceptChange(ChangeMode mode) {
 		if (!Block.class.isAssignableFrom(getExpr().getReturnType()))
 			return null;
 		if (mode == ChangeMode.SET)
 			return CollectionUtils.array(Direction.class);
 		return null;
 	}
 	
-	@SuppressWarnings("deprecation")
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) throws UnsupportedOperationException {
+	@SuppressWarnings("deprecation")
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) throws UnsupportedOperationException {
 		assert mode == ChangeMode.SET;
 		assert delta != null;
 		
-		final Block b = (Block) getExpr().getSingle(e);
+		Block b = (Block) getExpr().getSingle(e);
 		if (b == null)
 			return;
+      
 		BlockData data = b.getBlockData();
 		if (data instanceof org.bukkit.block.data.Directional) {
 			((org.bukkit.block.data.Directional) data).setFacing(toBlockFace(((Direction) delta[0]).getDirection(b)));
```

Comment:
A null check for delta[0] should be done (Not technically part of your PR, but something good to add)